[TM] ADD: Расширение системы РНД

Mirag1993 opened the pull request:
## Описание / Что этот PR делает
<img width="378" height="494" alt="image" src="https://github.com/user-attachments/assets/1d47056f-023f-47a8-a36f-b9a0d573fb47" />

Добавляет NtOS приложение "Cognitive Research Suite" которое представляет собой набор головоломок.
Предназначен для изучения логических паттернов и изучения когнитивных шаблонов сознания. При успешном прохождении головоломок начисляет очки исследований которые можно распечатать прямо в вашем устройстве.

Набор голололомок представляет собой:

**Lights out**

<img width="793" height="643" alt="image" src="https://github.com/user-attachments/assets/bd827e8f-b3b8-4ce7-8529-a816a71f0a69" />

**Matermind**

<img width="801" height="660" alt="image" src="https://github.com/user-attachments/assets/fb3499a1-2c62-407f-ace4-1c3145c66a41" />

**Cryptogram**

<img width="807" height="658" alt="image" src="https://github.com/user-attachments/assets/0304cabb-96e3-4cab-8f0f-0ec43e3cdc81" />

**Sudoku**

<img width="792" height="655" alt="image" src="https://github.com/user-attachments/assets/fa52feed-2d20-42b0-9dfa-bebf40021eb8" />

**Topsort**

<img width="794" height="658" alt="image" src="https://github.com/user-attachments/assets/58cd3e80-9862-49ae-b37f-eee702540f9a" />

## Причина создания ПР / Почему это хорошо для игры
Позволяет разнообразить исследования и получить альтернативный источник получения научных знаний.
## Демонстрация изменений / Тестирование
Установить приложение на любой носитель NtOS системы. Запустить приложение и пройти испытание. Получить очки исследований.
## Список изменений
Прошу ознакомится подробно с файлом README
```yml
🆑
add: NtOS:Cognitive Research Suite
balance: Добавлена альтернатива получения очков исследований
/🆑
```

Mirag1993 committed:
ADD: Расширение системы РНД

Добавляет NTOS-приложение Cognitive Research Suite: когнитивные мини‑симуляции с наградой RP.
TGUI-интерфейс, 5 режимов: Lights Out, Mastermind, Sudoku 4×4, Logic, Topsort
Счёт и выдача research notes; пер‑режимный кулдаун антиабуза
Доступ: RD или Captain; доступно в NT Software Hub; требует NTNet
Документация: mod_celadon/docs/CRS/README.md

Mirag1993 commented:
https://github.com/CeladonSS13/Shiptest/issues/2290 - частично закрывает идею

Mirag1993 committed:
Fix ai review

Fix ai review

Mirag1993 committed:
Fix 3

Fix 3 (AI review)

Mirag1993 committed:
Update (remove xor ; add crypto)

Mirag1993 committed:
Fix 4

Fix 4

MysticalFaceLesS commented:
<!-- tgs_test_merge_comment -->
## Test merge deployment history:

<details><summary>Test Merge Deployed @ 10/04/2025 15:08:29 +00:00:</summary>

##### Server Instance
Shiptest_Alpha

##### Merged By
De-Grand

##### Revision
Origin: ed76e50ecaefcd7cad8d7a757f2eb5edc81c3589
Pull Request: 8daea9246ea4a61095ad1eb6aaf86135689deff5
Server: 4a6892185ea771d2efacdce40a58569a614c8f41
</details>

Mirag1993 commented:
лайтс аут 5 на 5 не решаемая

Mirag1993 commented:
Вместо админской кнопки отображается ноль у пользователей

Mirag1993 commented:
Слишком быстро сбрасывается прогресс

Mirag1993 committed:
Fix na Fix

Уменьшена сложность лайтс аут до 6 на 6 , исправлено судоку , лайтс аут , кнопка дебага больше не пишется как нолик . Статистика не удаляется посреди раунда

Mirag1993 committed:
Merge branch 'CeladonSS13:beta-dev' into Research-Hub

Mirag1993 committed:
Updare critical bug fix

1.Фикс крит бага который мешал рестарту сервера
2.Заменили мастермайнд на более интерактивную игру

MysticalFaceLesS commented on mod_celadon/docs/CRS/README.md:
```diff
@@ -0,0 +1,212 @@
+## Cognitive Research Suite (CRS)
```
Удалить файл.

Mirag1993 commented:
я не вижу комментов войко но представим что они там есть. С переносом пра в модуль не согласен , никакой явной логики в этом нет кроме возможного конфликта одной строчки в основном файле билда. У меня нет возможности и нервов в данный момент допилить эти хотелки. Также не согласен с удалением файла ридми, но опятьже нет желания доказывать чтото. Пр закрывается на неопределенный. срок.
